core/src/main/java/io/grpc/Server.java

```diff
@@ -52,6 +52,18 @@
   public abstract Server start() throws IOException;
 
   /**
+   * Returns the port number the server is listening on.  This can return -1 if there is no actual
+   * port or the result otherwise does not make sense.  Result is undefined after the server is
+   * terminated.
+   *
+   * @return the port number.
+   */
+  @ExperimentalApi
+  public int getPort() {
```

Comment:
Because -1 is how URL reports an absent port number.  Precedent in the standard library, basically.